tools/autograd/gen_variable_type.py

```diff
@@ -961,6 +974,16 @@ def setup_derivative(differentiable_inputs):
         body.append(SETUP_DERIVATIVE.substitute(env, setup=setup))
         return body
 
+    def emit_check_if_in_complex_autograd_whitelist():
+        body = []
+        if base_name in GRADIENT_IMPLEMENTED_FOR_COMPLEX:
+            return body
+        for arg in differentiable_outputs:
+            if arg['type'] == 'Tensor':
```

Comment:
What about TensorList?
In particular functions like `unbind()` will return such objects.

Reply:
The functions that are differentiable and return TensorList are: `torch.unbind`, `torch.split` (both of which have correct backward definition for complex). So, I think its ok to just not do anything for that case. However, I'll add check for TensorList type for any functions that maybe added in future.

Reply:
There are no other? Sounds good.
But yes a check is nice to make sure we don't break this in the future.

Reply:
There's `_cudnn_rnn_backward` but it's non-differentiable. 
Added `split', split_with_sizes, unsafe_split, split_with_sizes_backward` to the list.